### PR TITLE
Teleterm: Define / Set AssumeStartTime fields

### PR DESCRIPTION
**Pull request by kimlisa**
part of https://github.com/gravitational/teleport/issues/35436

suggest reivewing by commit

This PR just defines and sets the `AssumeStartTime` fields in backend/frontend when: creating, reviewing, and listing

### CODE REVIEW

File: lib/teleterm/clusters/cluster_access_requests.go [hunk 1]
```diff
@@ -151,6 +152,15 @@ func (c *Cluster) CreateAccessRequest(ctx context.Context, req *api.CreateAccess
...
 	request.SetSuggestedReviewers(req.SuggestedReviewers)
 
+	if req.GetAssumeStartTime() != nil && !req.AssumeStartTime.AsTime().IsZero() {
+		assumeStartTime := req.AssumeStartTime.AsTime()
+		if time.Until(assumeStartTime) > constants.MaxAssumeStartDuration {
+			return nil, trace.BadParameter("assumeStartTime too far in future: latest date %q",
+				assumeStartTime.Add(constants.MaxAssumeStartDuration).Format(time.RFC3339))
+		}
+		request.SetAssumeStartTime(assumeStartTime)
+	}
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply lib/teleterm/clusters/cluster_access_requests.go]: i consulted the original author and it was an oversite. i added the validation in the backend, but maybe it's also good to keep the check to fail earlier? i don't have strong preference

File: web/packages/teleterm/src/services/tshd/createClient.ts [hunk 2]
```diff
@@ -300,6 +301,7 @@ export function createTshdClient(
         suggestedReviewers: params.suggestedReviewers,
         roles: params.roles,
         reason: params.reason,
+        assumeStartTime: getProtoTimestamp(params.assumeStartTime),
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply web/packages/teleterm/src/services/tshd/createClient.ts:304]: oh nice! thanks 👍

File: lib/teleterm/clusters/cluster_access_requests.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (gzdunek) [reply lib/teleterm/clusters/cluster_access_requests.go]: I'd keep only the server-side validation, having it in a single place is easier to maintain. 
The user doesn't have any benefit from that client side check.

[the remark this answers is not in the document]
Comment (kimlisa) [reply lib/teleterm/clusters/cluster_access_requests.go]: removed

File: lib/teleterm/apiserver/handler/time_converter.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import (
+	"time"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func getProtoTimestamp(time *time.Time) *timestamppb.Timestamp {
+	var protoTime *timestamppb.Timestamp
+
+	if time != nil && !time.IsZero() {
```
[the remark this answers is not in the document]
Comment (kimlisa) [reply lib/teleterm/apiserver/handler/time_converter.go]: i think it started with sometimes this field is set as a pointer

File: proto/teleport/lib/teleterm/v1/access_request.proto [hunk 4]
```diff
@@ -62,6 +64,10 @@ message AccessRequestReview {
   // promoted_access_list_title is the title of the access
   // list that the access request was promoted to.
   string promoted_access_list_title = 6;
+  // assume_start_time is the time the requested resources can be assumed.
+  // can overwrite the start time requested by the requester
+  // by the reviewer.
```
Comment (gzdunek) [inline proto/teleport/lib/teleterm/v1/access_request.proto:69]:
```suggestion 
  // The reviewer can overwrite the requested time.
```

**Review by gzdunek — APPROVED** [review]

File: lib/services/access_request.go
Comment (ravicious) [inline lib/services/access_request.go:1]: Even though this change is not that big, I feel like it deserves its own PR rather than being bundled together with teleterm changes. I think backend changes deserve more scrutiny by the virtue of being used by the server vs teleterm code which is merely a client.

File: lib/teleterm/clusters/cluster_access_requests.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ravicious) [reply lib/teleterm/clusters/cluster_access_requests.go]: Is there a situation in which the client can connect to a server which does not perform this validation? What should happen then?

I'd like to establish whether it's a problem for us if we let it through or not.

File: lib/services/access_request.go
Comment (ravicious) [reply lib/services/access_request.go:1]: Oh, I noticed that we're also moving the validation from `tool/` – this would be another argument in favor of preparing a separate PR.

**Comment by kimlisa** [comment]
closing in favor of https://github.com/gravitational/teleport/pull/39050, which extracts just defining new fields to teleterm api
